### PR TITLE
compact, receive, store: Init all labeled counter and histogram metrics

### LINKED ISSUE #2889: Init all labeled counter metrics to 0
**Thanos, Prometheus and Golang version used**:
thanos v0.13.0

**What happened**:
Thanos jobs counter metrics with labels are not initialized to 0 for the known labels.

For example, `thanos_compact_downsample_total{group="3600000@4233639831006398026"}` would not be exported by thanos-compact until the first 1h downsample after starting the server.  Prometheus queries are hard to manage with data series that disappear, it causes problems with `increase()`, etc.

**What you expected to happen**:
Per prometheus best practice, when thanos jobs start they should register the known label sets of counters so that the collector publishes a 0 value for the counters.

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixes #2889

### CODE REVIEW

**Comment by yeya24** [comment]
I am not sure about this one https://github.com/thanos-io/thanos/blob/master/pkg/store/bucket.go#L198-L201. Why this is a counter metric, not a histogram or summary metric?

**Comment by yeya24** [comment]
@pstibrany Hello, for this metric https://github.com/thanos-io/thanos/blob/master/pkg/store/bucket.go#L198-L201, is it intended to be a counter or a histogram? It doesn't have `_total` suffix

**Comment by pstibrany** [comment]
> @pstibrany Hello, for this metric https://github.com/thanos-io/thanos/blob/master/pkg/store/bucket.go#L198-L201, is it intended to be a counter or a histogram? It doesn't have `_total` suffix

Hi, it’s supposed to be counter and missing `_total` suffix is oversight :-(

**Comment by yeya24** [comment]
> > @pstibrany Hello, for this metric https://github.com/thanos-io/thanos/blob/master/pkg/store/bucket.go#L198-L201, is it intended to be a counter or a histogram? It doesn't have `_total` suffix
> 
> Hi, it’s supposed to be counter and missing `_total` suffix is oversight :-(

I can fix the name in another commit this pr if you don't mind.

**Comment by pstibrany** [comment]
> I can fix the name in another commit this pr if you don't mind.

That would be great, thanks a lot!

**Comment by yeya24** [comment]
> > I can fix the name in another commit this pr if you don't mind.
> 
> That would be great, thanks a lot!

Done. Note that update is also needed on the Cortex side. https://github.com/cortexproject/cortex/blob/aa0d48adf6309a406e34b2d9e467787328312efb/pkg/storegateway/bucket_store_metrics.go#L111

**Comment by pracucci** [comment]
> > > I can fix the name in another commit this pr if you don't mind.
> > 
> > 
> > That would be great, thanks a lot!
> 
> Done. Note that update is also needed on the Cortex side. https://github.com/cortexproject/cortex/blob/aa0d48adf6309a406e34b2d9e467787328312efb/pkg/storegateway/bucket_store_metrics.go#L111 😄

Definitely! We'll address it the next Thanos update in Cortex, once this PR is merged.

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -303,6 +303,12 @@ func runCompact(
 			if err := sy.SyncMetas(ctx); err != nil {
 				return errors.Wrap(err, "sync before first pass of downsampling")
 			}
+
+			for _, meta := range sy.Metas() {
+				groupKey := compact.DefaultGroupKey(meta.Thanos)
+				downsampleMetrics.downsamples.WithLabelValues(groupKey)
+				downsampleMetrics.downsamples.WithLabelValues(groupKey)
```
Comment (pracucci) [inline cmd/thanos/compact.go:310]:
```suggestion
				downsampleMetrics.downsampleFailures.WithLabelValues(groupKey)
```

Comment (yeya24) [reply cmd/thanos/compact.go:310]: Done. Thanks for catching this.

Btw just found another bug here https://github.com/thanos-io/thanos/blob/master/cmd/thanos/downsample.go#L234.

**Pull request merged**